### ISSUE TITLE
DEV: `dev.py` missing PYTHONPATH when building doc

### ISSUE THREAD

**Issue by tupui** [scipy/scipy#15638]
I noticed that the PYTHONPATH was not set when building doc.

I need to do this for it to work
```
export PYTHONPATH=$PWD/build-install/lib/python3.10/site-packages
python dev.py -n -j2 --doc html-scipyorg
```

cc @Smit-create

**Comment by rgommers** [comment]
So what is broken is `python dev.py --doc`.

**Comment by tupui** [comment]
> So what is broken is `python dev.py --doc`.

I think it should check if SciPy is importable, if not it should try to add it to the PYTHONPATH. I actually see that other commands don't do this too (although written in the help). This was the behaviour with `runtests.py` no?

**Comment by Smit-create** [comment]
I believe this is the same issue as addressed here: https://github.com/scipy/scipy/pull/15440. We can use the same fix for docs too. That works for me:
```
$ python dev.py -n --doc
Trying to import scipy from development installed path at: /home/admin-pc/Smitlunagariya/scipy/build-install/lib/python3.10/site-packages
make: Entering directory '/home/admin-pc/Smitlunagariya/scipy/doc'
mkdir -p build/html build/doctrees
LANG=C "/home/admin-pc/anaconda3/envs/scipy-dev/bin/python" -msphinx -WT --keep-going -t scipyorg  -b html -d build/doctrees  "-j1" source build/html-scipyorg 
Running Sphinx v4.3.1
SciPy (VERSION 1.9.0.dev0+1511.225c99f)
making output directory... done

```

**Comment by rgommers** [comment]
That sounds right. Just to make sure @Smit-create, are you saying that it already works for you now, or you have a local fix to make it work?

**Comment by Smit-create** [comment]
> are you saying that it already works for you now

Yes, that works for me. Opening a PR for this fix.

**Comment by Smit-create** [comment]
> are you saying that it already works for you now

 Yes, that works for me with the same fix as in #15440. Opening a PR for this fix.
